### PR TITLE
BAN-2029 Add Apple News Image support

### PR DESCRIPTION
**Pull request by damiendillon**
Add image support to the apple news export feature

https://mydevex.atlassian.net/browse/BAN-2029

### CODE REVIEW

**Comment by EgleMekaite** [comment]
+1

File: spec/article_json/export/apple_news/elements/image_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      caption: caption,
+      float: float,
+      alt: alt
+    )
+  end
+  let(:float) { nil }
+  let(:caption) { [ArticleJSON::Elements::Text.new(content: 'Foo Bar')] }
+  let(:alt) { 'Text description' }
+
+  let(:expected_json) { {:URL=>"/foo/bar.jpg", :caption=>"Foo Bar", :role=>"image"} }
```
Comment (ozgun) [inline spec/article_json/export/apple_news/elements/image_spec.rb]: How about `{ URL: "/foo/bar.jpg", caption: "Foo Bar", role: "image" }` ?

**Comment by ozgun** [comment]
Looks good!

When I was checking Apple News API documentation, I came across this: "Image filenames should be properly encoded as URLs." I wonder if we need to encode image URLs. Maybe Google exports them as URL encoded, not sure though.

**Comment by damiendillon** [comment]
> Looks good!
> 
> When I was checking Apple News API documentation, I came across this: "Image filenames should be properly encoded as URLs." I wonder if we need to encode image URLs. Maybe Google exports them as URL encoded, not sure though.

image

The above example url was generated by dragging and dropping an image to a google doc.

**Comment by damiendillon** [comment]
However, maybe we should encode it anyway, as this is a requirement and the input may vary. If you think I should, I am happy to adjust the PR.

File: spec/article_json/export/apple_news/elements/image_spec.rb (hunk 1 above)
Comment (dsager) [reply spec/article_json/export/apple_news/elements/image_spec.rb]: Yep, we should follow the style we use elsewhere (also using single quotes):
```suggestion
      let(:expected_json) { {URL: '/foo/bar.jpg', caption: 'Foo Bar', role: 'image'} }
```

File: lib/article_json/export/apple_news/elements/image.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module ArticleJSON
+  module Export
+    module AppleNews
+      module Elements
+        class Image < Base
+          # Headline
```
Comment (dsager) [inline lib/article_json/export/apple_news/elements/image.rb]: :eyes:

File: lib/article_json/export/apple_news/elements/image.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+          # @return [Hash]
+          def export
+            {
+              role: "image",
+              URL: @element.source_url,
```
Comment (dsager) [inline lib/article_json/export/apple_news/elements/image.rb:11]: Don't we have to upload images to apple news? Or would that step happen afterwards?

Comment (ozgun) [reply lib/article_json/export/apple_news/elements/image.rb:11]: It will happen afterwards as far as I remember.

**Comment by ozgun** [comment]
We shouldn't encode already encoded URL.

File: lib/article_json/export/apple_news/elements/image.rb (hunk 2 above)
Comment (damiendillon) [reply lib/article_json/export/apple_news/elements/image.rb]: 😬

File: spec/article_json/export/apple_news/elements/image_spec.rb (hunk 1 above)
Comment (damiendillon) [reply spec/article_json/export/apple_news/elements/image_spec.rb]: Amended

File: lib/article_json/export/apple_news/elements/image.rb (hunk 2 above)
Comment (damiendillon) [reply lib/article_json/export/apple_news/elements/image.rb]: Amended

**Pull request merged**